runtime/doc/lua.txt

```diff
@@ -2847,6 +2847,24 @@ vim.keymap.set({mode}, {lhs}, {rhs}, {opts})                *vim.keymap.set()*
 ==============================================================================
 Lua module: vim.fs                                                    *vim.fs*
 
+vim.fs.abspath({path}, {opts})                              *vim.fs.abspath()*
+    Convert path to an absolute path. Does not check if the path exists,
+    resolve symlinks or hardlinks (including `.` and `..`), or expand
+    environment variables. If the path is already absolute, it is returned
+    unchanged.
+
+    Parameters: ~
+      • {path}  (`string`) Path
+      • {opts}  (`table?`) Optional keyword arguments:
+                • {cwd}? (`string`, default: |current-directory|) Current
```

Comment:
Yeah I still don't get the point of this. I can't come up with a reason you'd need this. If you need to know the absolute path from a file that isn't relative to the current working directory, does that not mean that you already know the absolute path of the file as it would be `cwd + path`? If so, why can't this just be done with string concatenation?

Reply:
> Yeah I still don't get the point of this. I can't come up with a reason you'd need this. If you need to know the absolute path from a file that isn't relative to the current working directory, does that not mean that you already know the absolute path of the file as it would be `cwd + path`? If so, why can't this just be done with string concatenation?

Not necessarily. Assume you have a `cwd`, but you don't know if `path` is absolute or relative. Doing `cwd + path` won't work if `path` is already absolute. In which case, you'd first have to do a check if `path` is absolute by doing `abspath(path) == path`, and _then_ do `cwd + path` if path is not absolute, which imo is way too complicated to expect the average Neovim user to understand.

Reply:
> Assume you have a `cwd`

I already have a problem with this premise. Why would I have a random `cwd` that isn't the current working directory?

Reply:
> > Assume you have a `cwd`
> 
> I already have a problem with this premise. Why would I have a random `cwd` that isn't the current working directory?

I'll give you a usecase that I recently had, where I made a `Rename` command that allowed me to rename any file, and the new name would actually be a path that's relative to the directory of the file being renamed. Another usecase I can think of would be a fuzzy file searcher that allows you to specify a directory, then shows you the filenames relative to the target directory you specified.

I'm sure @echasnovski can think of other usecases as well.

Reply:
> I'll give you a usecase that I recently had, where I made a Rename command that allowed me to rename any file, and the new name would actually be a path that's relative to the directory of the file being renamed.

Can I take a look at the code? Could you link it?

Reply:
> > I'll give you a usecase that I recently had, where I made a Rename command that allowed me to rename any file, and the new name would actually be a path that's relative to the directory of the file being renamed.
> 
> Can I take a look at the code? Could you link it?

https://github.com/famiu/dot-nvim/blob/master/plugin/file_cmds.lua

Reply:
> I'm sure @echasnovski can think of other usecases as well, considering that he agreed with having `cwd` in Matrix.

Sure. One of the real examples for me was to get absolute paths of particular subdirectories for all directories in 'runtimepath'. Here we have "many `cwd`" + "single `path`".

Now, this particular case (and many more) can be solved with `vim.fs.joinpath(vim.fs.abspath(cwd), path)`, but it is not 100% the case due to some obscure Windwos paths. So having `opts.cwd` seems logical to me.

Reply:
> > I'm sure @echasnovski can think of other usecases as well, considering that he agreed with having `cwd` in Matrix.
> 
> Sure. One of the real examples for me was to get absolute paths of particular subdirectories for all directories in 'runtimepath'. Here we have "many `cwd`" + "single `path`".
> 
> Now, this particular case (and many more) can be solved with `vim.fs.joinpath(vim.fs.abspath(cwd), path)`, but it is not 100% the case due to some obscure Windwos paths. So having `opts.cwd` seems logical to me.

It also just adds 3 lines of code and is very low-cost, so basically not much reason to not have it.

Reply:
Specifying a separate `cwd` is fine. This is similar to the `openat` function. It's not a common use case, but there's no reason not to support it (especially if it's easy to support, as @famiu said).